locale/en/docs/guides/security/index.md

```diff
@@ -0,0 +1,451 @@
+of the dependencies (allowing for unwanted updates) and/or common typos in the
+specification (vulnerable to [typosquatting][]).
+
+An attacker that takes control of an upstream package can publish a new version
+with malicious code in it. If a Node.js application depends on that package
+without being strict on which version is safe to use, the package can be
+automatically updated to the latest malicious version, compromising the application.
+
+Dependencies specified in the `package.json` file can have an exact version number
+or a range. However, when pinning a dependency to an exact version, its
+transitive dependencies are not themselves pinned.
+This still leaves the application vulnerable to unwanted/unexpected updates.
+
+Possible vector attacks:
+
+* Typosquatting attacks
+* Lockfile poisoning
+* Compromised maintainers
+* Malicious Packages
+* Dependency Confusions
+
+**Mitigations**
+
+* Prevent npm from executing arbitrary scripts with `--ignore-scripts`
+  * Additionally, you can disable it globally with `npm config set ignore-scripts true`
+* Pin dependency versions to a specific immutable version,
+  not a version that is a range or from a mutable source.
+* Use lockfiles, which pin every dependency (direct and transitive).
+  * Use [Mitigations for lockfile poisoning][].
+* Automate checks for new vulnerabilities using CI, with tools like [`npm-audit`][].
+  * Tools such as [`Socket`][] can be used to analyze packages with static analysis
+  to find risky behaviors such as network or filesystem access.
+* Use [`npm ci`][] instead of `npm install`.
+This enforces the lockfile so that inconsistencies between it and the
+_package.json_ file cause an error (instead of silently ignoring the lockfile
+in favor of _package.json_).
+* Carefully check the _package.json_ file for errors/typos in the names of the
+dependencies.
+
+### Memory Access Violation (CWE-284)
+
+Memory-based or heap-based attacks depends on a combination of memory management
+errors and an exploitable memory allocator.
+Like all runtimes, Node.js is vulnerable to these attacks if your projects runs
+on a shared machine.
+Using a secure heap is useful for preventing sensitive information from leaking
+due to pointer overruns and underruns.
+
+Unfortunately, secure heap is not available on Windows.
+More information can be found on Node.js [secure-heap documentation][].
+
+**Mitigations**
+
+* Use `--secure-heap=n` depending on your application where _n_ is the allocated
+maximum byte size.
+* Do not run your production app in a shared machine.
+
+### Monkey Patching (CWE-349)
+
+Monkey patching refers to the modification of properties in runtime aiming to
+change the existing behavior. Example:
+
+```js
+// eslint-disable-next-line no-extend-native
+Array.prototype.push = function (item) {
+  // overriding the global [].push
+};
+```
+
+This is **not** considered a vulnerability, it falls under the following
+statement: _“Node.js trusts the code being run”_.
```

Comment:
```suggestion
```

I'm suggesting removing a lot of these statements because they seem unnecessary and kind of defensive in a document like this.